recipes/cad1/task1/baseline/evaluate.py

```diff
@@ -0,0 +1,227 @@
+import csv
+import hashlib
+import itertools
+import json
+import logging
+from pathlib import Path
+from typing import Any, Dict, List, Tuple
+
+import hydra
+import numpy as np
+import pandas as pd
+from omegaconf import DictConfig
+from scipy.io import wavfile
+
+from clarity.evaluator.haaqi import haaqi_v1
+
+logger = logging.getLogger(__name__)
+
+
+class ResultsFile:
+    """A utility class for writing results to a CSV file.
+
+    Attributes:
+        file_name (str): The name of the file to write results to.
+    """
+
+    def __init__(self, file_name: str):
+        """Initialize the ResultsFile instance.
+
+        Args:
+            file_name (str): The name of the file to write results to.
+        """
+        self.file_name = file_name
+
+    def write_header(self):
+        """Write the header row to the CSV file."""
+        with open(self.file_name, "w", encoding="utf-8", newline="") as csv_file:
+            csv_writer = csv.writer(
+                csv_file, delimiter=",", quotechar='"', quoting=csv.QUOTE_MINIMAL
+            )
+            csv_writer.writerow(
+                [
+                    "song",
+                    "listener",
+                    "score",
+                    "l_bass",
+                    "r_bass",
+                    "l_drums",
+                    "r_drums",
+                    "l_other",
+                    "r_other",
+                    "l_vocals",
+                    "r_vocals",
+                ]
+            )
+
+    def add_result(
+        self,
+        listener: str,
+        song: str,
+        score: float,
+        instruments_scores: Dict[str, float],
+    ):
+        """Add a result to the CSV file.
+
+        Args:
+            listener (str): The name of the listener who submitted the result.
+            song (str): The name of the song that the result is for.
+            score (float): The combined score for the result.
+            instruments_scores (dict): A dictionary of scores for each instrument channel in the result.
+        """
+        logger.info(f"The combined score is {score}")
+
+        with open(self.file_name, "a", encoding="utf-8", newline="") as csv_file:
+            csv_writer = csv.writer(
+                csv_file, delimiter=",", quotechar='"', quoting=csv.QUOTE_MINIMAL
+            )
+            csv_writer.writerow(
+                [
+                    song,
+                    listener,
+                    str(score),
+                    str(instruments_scores["l_bass"]),
+                    str(instruments_scores["r_bass"]),
+                    str(instruments_scores["l_drums"]),
+                    str(instruments_scores["r_drums"]),
+                    str(instruments_scores["l_other"]),
+                    str(instruments_scores["r_other"]),
+                    str(instruments_scores["l_vocals"]),
+                    str(instruments_scores["r_vocals"]),
+                ]
+            )
+
+
+def compute_haaqi(
```

Comment:
Why not `from clarity.evaluator.haaqi.haaqi import haaqi_v1`?

The function looks very different there though, if this is a `haaqi_v2` it should be defined in the above module so it can be reused again.